packages/api/src/services/user.ts

```diff
@@ -0,0 +1,111 @@
+import { v4 as uuidv4 } from 'uuid';
+import { dynamoDbClient } from '@sf/api/config';
+import { PageInfo, PaginatedResponse, paginateResponse } from '@sf/api/utils/pagination';
+import {
+  CreateUserInput,
+  UpdateUserInput,
+  User,
+  UserFilter,
+} from '@sf/core/user';
+import { publishUser } from '@sf/api/subscriptions';
+import { validateDob } from '@sf/api/utils/validators';
+
+export const TableName = 'Users';
+
+export interface UserService {
+  userMatches(user: User, filter?: UserFilter): boolean;
+  findAll(pageInfo: PageInfo, filter?: UserFilter): Promise<PaginatedResponse<User>>;
+  findOne(id: string): Promise<User>;
+  create(input: CreateUserInput): Promise<User>;
+  update(input: UpdateUserInput): Promise<User>;
+  delete(id: string): Promise<void>;
+}
+
+const generateImageUrl = (name: string): string =>
+  `https://i.pravatar.cc/150?u=${encodeURIComponent(name)}`;
+
+export const userService: UserService = new (class implements UserService {
+  userMatches(user: User, filter?: UserFilter): boolean {
+    if (!filter) { return true; }
+    if (!filter.name) { return true; }
+    const regex = new RegExp(`${filter.name}`, 'i');
+    return regex.test(user.name);
+  }
+
+  async findAll(pageInfo: PageInfo, filter?: UserFilter) {
+    const { Items } = await dynamoDbClient.scan({ TableName }).promise();
+    const users = Items as User[];
+    const filtered = users.filter(user => this.userMatches(user, filter));
+    return paginateResponse(filtered, pageInfo);
+  }
+
+  async findOne(id: string) {
+    const response = await dynamoDbClient.get({ TableName, Key: { id } }).promise();
+    const user = response.Item as User | undefined;
+    if (!user) {
+      console.error(`Invalid user id: ${id}`);
```

Comment:
What are your thoughts on `console.log` over logging libraries?

Reply:
I am a fan of `winston-cloudwatch` as I have been using them On-Premise servers as well.
`node-lambda-log` is also what I have used in one of my recent project and it seems stable as well.
WE can log them into `CloudWatch` and add some pattern filter to trigger SNS for notifications for critical errors, ...